### ISSUE TITLE
Payments Block: Stripe's Test Mode is not recognized by payment block

### ISSUE THREAD

**Issue by supernovia** [Automattic/wp-calypso#96704]
### Quick summary

Users who want to test their purchase process can't, because even though Stripe is set to Test Mode, WordPress.com doesn't recognize the mode and instead returns an error:

> "Your card was declined. Your request was in live mode, but used a known test card"

### Steps to reproduce

1. Connect stripe and create a payments block
2. Set Stripe to "Test Mode"
3. Try making a test purchase

### What you expected to happen

Users should be able to make test purchases before going live

### What actually happened

Users can't make test purchases, and instead have to resort to making and refunding small charges -- but then they lose processing fees.

### Impact

Some (< 50%)

### Available workarounds?

No and the platform is unusable

### If the above answer is "Yes...", outline the workaround.

Marking "unusable" because users shouldn't have to pay processing fees to test.

### Logs or notes

This problem pops up periodically and confuses HEs and users. It would be nice if test mode just worked.

**Comment by supernovia** [comment]
9082796-zd-a8c - we've seen other cases in the past, too.

**Comment by jartes** [comment]
📌 REPRODUCTION RESULTS
- Replicated

📌 ACTIONS
- Triaged
- Switching from `Bug` to `Feature request`, as this is not something we have implemented in Dotcom.
- Lowering priority from `Blocker` to `Normal`. 

@Automattic/shilling is this something that you already considered previously? Thank you!

**Comment by sirbrillig** [comment]
> is this something that you already considered previously? Thank you!

Good question! Since this is about Payment Blocks and not our MOR checkout I believe this is a question for @Automattic/gold.

**Comment by millerf** [comment]
@supernovia Yes, this is something that was talked about with @Automattic/zap at some point.
